### ISSUE TITLE
Apple's Accelerate framework support

### ISSUE THREAD

**Issue by romanbsd** [xiph/speexdsp#25]
I ported all of the CPU intensive parts of the echo cancellation and preprocess to use the Apple's Accelerate framework. If it's of any interest, I can prepare and submit a PR.

**Comment by yycking** [comment]
@romanbsd
Can you share these code with me?

**Comment by romanbsd** [comment]
Sorry, missed this comment. I think most (if not all) of it is here:
https://github.com/romanbsd/libspeexdsp
If something is missing, ping me in a week from now, since I'm abroad, and the files are on another computer.

**Comment by tmatth** [comment]
@romanbsd do you have any benchmarks?

**Comment by romanbsd** [comment]
No, don't have it handy, was too long ago. But it was at least 4x faster as far as I remember. Also I recall that there were places where I tried rewriting in order to use accelerated functions, and it wasn't worth it, so if you see places that are possible candidates - chances are that I already tried

**Comment by tmatth** [comment]
> No, don't have it handy, was too long ago. But it was at least 4x faster as far as I remember. Also I recall that there were places where I tried rewriting in order to use accelerated functions, and it wasn't worth it, so if you see places that are possible candidates - chances are that I already tried

Thanks I'll try and take a look, sorry this got missed (this github project is just a mirror of https://gitlab.xiph.org/xiph/speexdsp so PRs sent here can get overlooked).

**Comment by romanbsd** [comment]
Note that this is production quality code. It runs in production on Android app for some years now.

**Comment by romanbsd** [comment]
P.S. The changes are in the mdf.c and filterbank.c
